### PR TITLE
Preserve existing pod-preset certs

### PR DESCRIPTION
**Pull request by kwiatekus**
**Related issue**
https://github.com/kyma-project/kyma/issues/12531
Requires kyma charts changes - https://github.com/kyma-project/kyma/pull/12540

### CODE REVIEW

**Comment by kwiatekus** [comment]
Test it with:
```
./bin/mothership-darwin local --components cluster-essentials  --kubeconfig {path to your kubeconfig}  --version PR-12540
```

File: .gitignore [hunk 1]
```diff
@@ -76,3 +76,4 @@ out.txt
 
 # ignore workspace sub-directory
 .workspace
+pkup.sh
```
Comment (tobiscr) [inline .gitignore:79]: what is that? :D

Comment (tobiscr) [inline .gitignore:79]:
```suggestion
```

**Review by tobiscr — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (kwiatekus) [reply .gitignore]: removed

**Review by moelsayed — COMMENTED** [review]
Looks great, just a couple for minor comments.

File: pkg/reconciler/instances/clusteressentials/action.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	mutatingWebhookConfiguration, err := k8sClient.AdmissionregistrationV1().MutatingWebhookConfigurations().Get(context.Context, mutatingWebhookConfigName, metav1.GetOptions{})
+	if err != nil {
+		logger.Errorf("Error while fetching existing mutatingWebhookConfigurations [%s]... Certificates will be re-generated", err.Error())
+	} else if mutatingWebhookConfiguration != nil {
+		webhooks := mutatingWebhookConfiguration.Webhooks
+		for _, w := range webhooks {
+			if w.Name == webhookname {
+				if w.ClientConfig.CABundle != nil {
+					logger.Infof("MutatingWebhookConfiguration [%s/%s] found. Attempting to reusing existing CA bundle", mutatingWebhookConfigName, webhookname)
+					context.Task.Configuration["pod-preset.caCert"] = string(w.ClientConfig.CABundle)
```
[the remark this answers is not in the document]
Comment (kwiatekus) [reply pkg/reconciler/instances/clusteressentials/action.go:38]: yup
https://github.com/kyma-incubator/reconciler/blob/c79cf22addc5bba86c16bd59c5366e7f76eb6cc0/cmd/reconciler/start/service/http.go#L68

File: pkg/reconciler/instances/clusteressentials/action.go [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+	logger.Infof("Action '%s' executed (passed version was '%s')", a.name, context.Task.Version)
+
+	mutatingWebhookConfiguration, err := k8sClient.AdmissionregistrationV1().MutatingWebhookConfigurations().Get(context.Context, mutatingWebhookConfigName, metav1.GetOptions{})
+	if err != nil {
+		logger.Errorf("Error while fetching existing mutatingWebhookConfigurations [%s]... Certificates will be re-generated", err.Error())
+	} else if mutatingWebhookConfiguration != nil {
+		webhooks := mutatingWebhookConfiguration.Webhooks
+		for _, w := range webhooks {
+			if w.Name == webhookname {
+				if w.ClientConfig.CABundle != nil {
```
[the remark this answers is not in the document]
Comment (kwiatekus) [reply pkg/reconciler/instances/clusteressentials/action.go]: yes. thx. done

File: pkg/reconciler/instances/utils/utils.go [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+func SetOverrideFromSecret(logger *zap.SugaredLogger, secret *v1.Secret, configuration map[string]interface{}, secretKey string, overridePath string) {
+	secretValue, err := ReadSecretKey(secret, secretKey)
+	if err != nil {
+		logger.Errorf("Error while fetching %s from secret... Override for path [%s] will be generated : [%s]", secretKey, overridePath, err.Error())
+		return
+	}
+	if secretValue != "" {
+		if secretValue == "true" || secretValue == "false" {
+			configuration[overridePath], _ = strconv.ParseBool(secretValue)
```
[the remark this answers is not in the document]
Comment (kwiatekus) [reply pkg/reconciler/instances/utils/utils.go]: done

**Review by tobiscr — DISMISSED** [review]

**Review by rJankowski93 — APPROVED** [review]

**Pull request merged**